Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm

```diff
@@ -1415,3 +1415,191 @@ - (void)webView:(WKWebView *)webView stopURLSchemeTask:(id <WKURLSchemeTask>)tas
     }];
     TestWebKitAPI::Util::run(&done);
 }
+
+TEST(ResourceLoadStatistics, StorageAccessPromptSiteWithQuirk)
```

Comment:
Without reading all the details in this test, does it also test negatively? I.e. does it test that storage is not granted to a domain that's not in the set?